runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +162,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TransactionAbortedException) {
```

Comment:
Ok I have a question/comment around this.
1. Can we not unify how we handle trimmed exception ? Or is the handling for nontransactional case very different? (Hopefully we are not missing anything in that case) It is ok to use the same implementation at different places.